### PR TITLE
[Refactor] Improve MediaRenderer component and update token components

### PR DESCRIPTION
**Pull request by jnsdls**
## PR-Codex overview
This PR focuses on enhancing ERC20 functionality and improving token handling in the dashboard.

### Detailed summary
- Added `transferBatch()` extension to `erc20.ts`
- Updated `TokenDetails` to include `chainId`
- Enhanced token supply handling in `TokenSupplyLayout`
- Improved token contract detection in `ContractTokensPage`
- Added `transferBatch` function for batch ERC20 token transfers

> The following files were skipped due to too many changes: `apps/dashboard/src/contract-ui/tabs/tokens/components/airdrop-button.tsx`, `apps/dashboard/src/contract-ui/tabs/tokens/components/transfer-button.tsx`, `apps/dashboard/src/contract-ui/tabs/tokens/components/supply.tsx`, `apps/dashboard/src/contract-ui/tabs/tokens/components/airdrop-form.tsx`, `apps/dashboard/src/contract-ui/tabs/tokens/components/transfer-form.tsx`, `apps/dashboard/src/contract-ui/tabs/tokens/components/burn-form.tsx`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by jnsdls** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3602?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#3602** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3602?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#3601** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3601?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @jnsdls and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/dashboard/src/contract-ui/tabs/tokens/components/burn-form.tsx [hunk 1]
```diff
@@ -81,40 +83,48 @@ export const TokenBurnForm: React.FC<TokenBurnFormProps> = ({ contract }) => {
...
-                    trackEvent({
-                      category: "token",
-                      action: "burn",
-                      label: "error",
-                      error,
-                    });
-                    onError(error);
-                  },
+
+              // TODO: burn should be updated to take amount / amountWei (v6?)
```
Comment (joaquim-verges) [inline apps/dashboard/src/contract-ui/tabs/tokens/components/burn-form.tsx:98]: Ughhh

File: apps/dashboard/src/contract-ui/tabs/tokens/page.tsx [hunk 2]
```diff
@@ -16,13 +17,23 @@ export const ContractTokensPage: React.FC<ContractTokenPageProps> = ({
...
+    !contractQuery?.contract ||
+    !contractAddress ||
+    !chainId
+  ) {
     // TODO build a skeleton for this
     return <div>Loading...</div>;
   }
 
-  if (!contractQuery?.contract) {
+  const isERC20 = detectFeatures<TokenContract>(contractQuery.contract, [
```
Comment (joaquim-verges) [inline apps/dashboard/src/contract-ui/tabs/tokens/page.tsx:32]: That's still V4 underneath right?

File: apps/dashboard/src/hooks/useTxNotifications.ts [hunk 3]
```diff
@@ -24,10 +35,19 @@ export function useTxNotifications(
       isClosable: true,
     });
     if (contract) {
-      invalidateContractQuery({
-        chainId: contract.chainId,
-        contractAddress: contract.getAddress(),
-      });
+      if ("getAddress" in contract) {
```
Comment (joaquim-verges) [inline apps/dashboard/src/hooks/useTxNotifications.ts:38]: Can't wait to clean all of this up

File: packages/thirdweb/src/extensions/erc20/write/transferBatch.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+ * import { transferBatch } from "thirdweb/extensions/erc20";
+ * import { sendTransaction } from "thirdweb";
+ *
+ * const transaction = transferBatch({
+ *  contract,
+ *  batch: [
+ *    {
+ *      to: "0x...",
+ *      amount: 100,
+ *    },
```
Comment (joaquim-verges) [inline packages/thirdweb/src/extensions/erc20/write/transferBatch.ts:38]: Maybe add another recipient in the example?

**Review by joaquim-verges — APPROVED** [review]

File: apps/dashboard/src/contract-ui/tabs/tokens/components/burn-form.tsx (hunk 1 above)
Comment (jnsdls) [reply apps/dashboard/src/contract-ui/tabs/tokens/components/burn-form.tsx:98]: [deadcode](https://app.graphite.dev//settings/meme-library?org=thirdweb-dev)

File: apps/dashboard/src/contract-ui/tabs/tokens/page.tsx (hunk 2 above)
Comment (jnsdls) [reply apps/dashboard/src/contract-ui/tabs/tokens/page.tsx:32]: yes, sadly

File: packages/thirdweb/src/extensions/erc20/write/transferBatch.ts (hunk 4 above)
Comment (jnsdls) [reply packages/thirdweb/src/extensions/erc20/write/transferBatch.ts:38]: mhk

**Pull request merged**